client/src/components/Profile/Profile.jsx

```diff
@@ -72,32 +69,31 @@ function Profile() {
   // upload image
   async function uploadImage(event) {
     event.persist();
-    try {
-      //set loading sign
-      setLoadingImg(true);
-
-      const isList = await listObjects(email)
-      await isList.map(elem => ({ "Key": elem.Key }))
-      .then(result => deleteObjects(email, result));
-
     
-      await saveObject(email, event.target.files[0]);
-      const newImg = await getSingleObject(
-        email,
-        `${email}/${event.target.files[0].name}`
-      );
+    // set loading sign
+    setLoadingImg(true);
 
-      setImage(newImg);
-    } catch {
+    const fileInfo = await listObjects(email);
+    const imageKey = fileInfo.map(elem => ({ "Key": elem.Key }) );
+
+    if (imageKey.length === 0) {
       await saveObject(email, event.target.files[0]);
       const newImg = await getSingleObject(
         email,
         `${email}/${event.target.files[0].name}`
       );
       setImage(newImg);
-      //close loading sign
       setLoadingImg(false);
```

Comment:
do you want to `return` at this point, or do you want the function to continue?

Reply:
return